src/modules/jwt-encryption-service/JwtEncryptionService.ts

```diff
@@ -0,0 +1,49 @@
+import 'reflect-metadata';
+
+import { randomBytes } from 'crypto';
+import { JWE, JWK } from 'node-jose';
+
+import CompanyAuthConfig from 'app/models/CompanyAuthConfig';
+
+interface AuthPayload {
+    nonce: string;
+    content: string;
+}
+
+export default class JwtEncryptionService {
+    public constructor(private companyAuthConfig: CompanyAuthConfig) {}
+
+    public generateNonce(): string {
+        const bytes = randomBytes(5);
+        const buffer = Buffer.from(bytes);
+        return buffer.toString('base64');
+    }
+
+    public async jweEncodeWithNonce(returnUri: string, nonce: string): Promise<string> {
```

Comment:
I would consider rename as for me it is not encoding process, it is encrypting process.